docs/sw2sphinxopenapi.py

```diff
@@ -0,0 +1,142 @@
+"""Generate Markdown documents from an OpenAPI swagger file."""
+
+from __future__ import print_function
+
+import contextlib
+import os
+import os.path
+import re
+import sys
+import textwrap
+
+import yaml
+
+
+class OutputFiles(object):
+    """A context manager to manage a series of files.
+
+    Use like this::
+
+        with OutputFiles() as outfiles:
+            ...
+            if some_condition():
+                f = outfiles.open("filename.txt", "w")
+
+    Each open will close the previously opened file, and the end of the with
+    statement will close the last one.
+
+    """
+    def __init__(self):
+        self.file = None
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        if self.file:
+            self.file.close()
+        return False
+
+    def open(self, *args, **kwargs):
+        if self.file:
+            self.file.close()
+        self.file = open(*args, **kwargs)
+        return self.file
+
+
+@contextlib.contextmanager
+def redirect_output(file):
+    old_stdout = sys.stdout
+    try:
+        sys.stdout = file
+        yield
+    finally:
+        sys.stdout = old_stdout
+
+
+sluggers = [
+    r"^.*?/v\d+/[\w_-]+",
+    r"^(/[\w_-]+){,3}",
+]
+
+method_order = ['get', 'post', 'put', 'patch', 'delete', 'head', 'options']
+
+
+def method_ordered_items(method_data):
+    keys = [k for k in method_order if k in method_data]
+    for key in keys:
+        yield key, method_data[key]
```

Comment:
This is a personal opinion, but I find this piece of code clearer:

```
for k in keys:
    if key in method_data:
        yield key, method_data[key]
```

Reply:
Yes, sorry, I meant to write:

```
for k in method_order:
    if key in method_data:
        yield key, method_data[key]
```

That way, I believe the order of keys is respected.